### ISSUE TITLE
Prevent `TranslationCreator` create `Translation` objects with the same locale as the source locale`

### ISSUE THREAD

**Issue by zerolab** [wagtail/wagtail-localize#749]
From an internal report where a site somehow ended up with multiple `Translation` objects where `target_locale` was the same as `soruce.locale` which the resulted in some odd behaviours.

Update https://github.com/wagtail/wagtail-localize/blob/main/wagtail_localize/operations.py#L63-L71 to skip any processing when `target_locale == source.locale`

**Comment by ACK1D** [comment]
Maybe this solution could help?
```python
		for target_locale in self.target_locales:
            # Skip if target_locale is the same as source locale
            if target_locale == source.locale:
                continue
```

**Comment by zerolab** [comment]
Yep, that's pretty much the idea

**Comment by zerolab** [comment]
Fixed by #756
